### PR TITLE
Add docker compose

### PR DESCRIPTION
**Pull request by sladyn98**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    # Configuration for building the docker image for the backend service
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "8080:8080" # Forward the exposed port 8080 on the container to port 8080 on the host machine
+    restart: always
+
+  app-client:
+    container_name: sample
```
Comment (LinuxSuRen) [inline docker-compose.yml:16]: Please consider to give it a better name instead of `sample`.

**Comment by sladyn98** [comment]
```
 npm ERR! code ENOENT
cds_front-end | npm ERR! syscall open
cds_front-end | npm ERR! path /app/package.json
cds_front-end | npm ERR! errno -2
cds_front-end | npm ERR! enoent ENOENT: no such file or directory, open '/app/package.json'
cds_front-end | npm ERR! enoent This is related to npm not being able to find a file.
cds_front-end | npm ERR! enoent 
```
The `docker compose up` command fails with this, even though the package.json is in the app folder I have no idea why the container cannot find it, maybe we just ditch the frontend docker file and build it out of the docker compose file itself something like this 
```
version: "2"
services:
  node:
    image: "node:8"
    user: "node"
    working_dir: /home/node/app
    environment:
      - NODE_ENV=production
    volumes:
      - ./:/home/node/app
    expose:
      - "8081"
    command: "npm start"
```

File: frontend/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 FROM node:13.12.0-alpine
 
 # set working directory
-WORKDIR /app
+WORKDIR /home/app
```
Comment (kwhetstone) [inline frontend/Dockerfile:5]: Replace the file like so and this should fix your docker-compose issue with the frontend.
```suggestion
# Start with a base image containing Java runtime
FROM openjdk:8-jdk-alpine

# Add Maintainer Info
LABEL maintainer="sladynnunes98@gmail.com"

# Add a volume pointing to /tmp
VOLUME /tmp

# Make port 8080 available to the world outside this container
EXPOSE 8080

# The application's jar file
ARG JAR_FILE=target/custom-distribution-service-0.0.1.jar

# Add the application's jar to the container
ADD ${JAR_FILE} custom-distribution-service.jar

# Run the jar file 
ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","/custom-distribution-service.jar"]
```

**Comment by sladyn98** [comment]
@kwhetstone It works perfect now, thanks :tada:

**Pull request merged**